src/sentry/api/endpoints/project_integration_details.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import absolute_import
+
+from sentry.api.bases.project import ProjectEndpoint, ProjectIntegrationsPermission
+from sentry.api.serializers import serialize
+from sentry.models import ProjectIntegration, Integration
+
+
+class ProjectIntegrationDetailsEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectIntegrationsPermission, )
+
+    def get(self, request, project, integration_id):
+        integration = ProjectIntegration.objects.get(
+            project=project,
+            integration_id=integration_id,
+        )
```

Comment:
same as above, avoid duplication - and use a `def get_object()` which `raise 404` if `Item.DoesNotExist` please.

Reply:
This would require more refactoring because this is not a Django view, and Rest Framework doesn't handle a normal `raise Http404`. Again, we already have precedent for this.

Reply:
>  and Rest Framework doesn't handle a normal raise Http404 

are you sure about that ? https://github.com/encode/django-rest-framework/blob/master/rest_framework/views.py#L71

anyway, not the point.

Reply:
@tonythomas01 we do not run off of DRF master and you need to raise a ResourceDoesNotExist for it to respond with an appropriate payload in Sentry.